test/unit/dao/QuadraticTimelock.test.ts

```diff
@@ -0,0 +1,335 @@
+import { expectEvent, expectRevert, getAddresses, getImpersonatedSigner, time } from '../../helpers';
+import { expect } from 'chai';
+import hre, { ethers } from 'hardhat';
+import { Signer } from 'ethers';
+
+const toBN = ethers.BigNumber.from;
+
+describe('QuadraticTimelockedDelegator', function () {
+  let userAddress;
+  let secondUserAddress;
+  let beneficiaryAddress1;
+
+  const impersonatedSigners: { [key: string]: Signer } = {};
+
+  before(async () => {
+    const addresses = await getAddresses();
+
+    // add any addresses you want to impersonate here
+    const impersonatedAddresses = [addresses.userAddress, addresses.secondUserAddress, addresses.beneficiaryAddress1];
+
+    for (const address of impersonatedAddresses) {
+      impersonatedSigners[address] = await getImpersonatedSigner(address);
+    }
+  });
+
+  beforeEach(async function () {
+    ({ userAddress, secondUserAddress, beneficiaryAddress1 } = await getAddresses());
+
+    this.tribe = await (await ethers.getContractFactory('MockTribe')).deploy();
+
+    this.window = toBN(4 * 365 * 24 * 60 * 60);
+    this.delegator = await (
+      await ethers.getContractFactory('QuadraticTimelockedDelegator')
+    ).deploy(this.tribe.address, userAddress, this.window);
+    this.totalTribe = toBN('10000');
+    await this.tribe.mint(this.delegator.address, this.totalTribe);
+  });
+
+  describe('Init', function () {
+    it('lockedToken', async function () {
+      expect(await this.delegator.lockedToken()).to.be.equal(this.tribe.address);
+    });
+
+    it('totalToken', async function () {
+      expect(await this.delegator.totalToken()).to.be.bignumber.equal(this.totalTribe);
+    });
+
+    it('should delegate voting power to beneficiary', async function () {
+      expect(await this.tribe.getCurrentVotes(userAddress)).to.be.bignumber.equal(this.totalTribe);
+    });
+  });
+
+  describe('Release', function () {
+    describe('Before cliff', function () {
+      it('reverts', async function () {
+        await time.increase((await this.delegator.cliffSeconds()).sub(toBN(1000)));
+        await expectRevert(this.delegator.release(userAddress, '100'), 'TokenTimelock: Cliff not passed');
+      });
+    });
+
+    describe('After cliff', function () {
+      it('releases tokens', async function () {
+        await time.increase((await this.delegator.cliffSeconds()).add(toBN(1)));
+        await this.delegator.release(userAddress, '1');
+        expect(await this.tribe.balanceOf(userAddress)).to.be.bignumber.equal(toBN('1'));
```

Comment:
Prefer getting starting TRIBE balance, then getting ending TRIBE balance and checking delta.